pkg/providers/amifamily/resolver.go

```diff
@@ -117,17 +119,20 @@ func New(amiProvider *Provider) *Resolver {
 // Multiple ResolvedTemplates are returned based on the instanceTypes passed in to support special AMIs for certain instance types like GPUs.
 func (r Resolver) Resolve(ctx context.Context, nodeTemplate *v1alpha1.AWSNodeTemplate, machine *v1alpha5.Machine, instanceTypes []*cloudprovider.InstanceType, options *Options) ([]*LaunchTemplate, error) {
 	amiFamily := GetAMIFamily(nodeTemplate.Spec.AMIFamily, options)
-	amis, err := r.amiProvider.Get(ctx, nodeTemplate, options)
+	mappedAMIs, err := r.getMappedAMIs(ctx, nodeTemplate, instanceTypes, options)
 	if err != nil {
 		return nil, err
 	}
-	if len(amis) == 0 {
-		return nil, fmt.Errorf("no amis exist given constraints")
+
+	kubeVersionStr, err := r.amiProvider.KubeServerVersion(ctx)
```

Comment:
`KubeServerVersion` should already have these structured values from within the `Discover().ServerVersion()` call. I wonder if we can just always return the structured version and then convert it to a string when it's needed elsewhere